File: src/models/Commande.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package models;

import java.util.Date;

/**
 *
 * @author choua
 */
public class Commande {
    
    private Double id, code_postal; 
    private Panier panier; 
    private double montant; 
    private String etat_commande, gouvernorat, ville, rue;
    private String date_commande; 

    public Commande() {
    }

    public Commande(Double id, Double code_postal, Panier panier, double montant, String etat_commande, String gouvernorat, String ville, String rue, String date_commande) {
        this.id = id;
        this.code_postal = code_postal;
        this.panier = panier;
        this.montant = montant;
        this.etat_commande = etat_commande;
        this.gouvernorat = gouvernorat;
        this.ville = ville;
        this.rue = rue;
        this.date_commande = date_commande;
    }

    public Commande(Double code_postal, Panier panier, double montant, String etat_commande, String gouvernorat, String ville, String rue, String date_commande) {
        this.code_postal = code_postal;
        this.panier = panier;
        this.montant = montant;
        this.etat_commande = etat_commande;
        this.gouvernorat = gouvernorat;
        this.ville = ville;
        this.rue = rue;
        this.date_commande = date_commande;
    }

    public Commande(Double code_postal, String gouvernorat, String ville, String rue) {
        this.code_postal = code_postal;
        this.gouvernorat = gouvernorat;
        this.ville = ville;
        this.rue = rue;
    }
    
    

    public Double getId() {
        return id;
    }

    public Double getCode_postal() {
        return code_postal;
    }

    public Panier getPanier() {
        return panier;
    }

    public double getMontant() {
        return montant;
    }

    public String getEtat_commande() {
        return etat_commande;
    }

    public String getGouvernorat() {
        return gouvernorat;
    }

    public String getVille() {
        return ville;
    }

    public String getRue() {
        return rue;
    }

    public String getDate_commande() {
        return date_commande;
    }

    public void setId(Double id) {
        this.id = id;
    }

    public void setCode_postal(Double code_postal) {
        this.code_postal = code_postal;
    }

    public void setPanier(Panier panier) {
        this.panier = panier;
    }

    public void setMontant(double montant) {
        this.montant = montant;
    }

    public void setEtat_commande(String etat_commande) {
        this.etat_commande = etat_commande;
    }

    public void setGouvernorat(String gouvernorat) {
        this.gouvernorat = gouvernorat;
    }

    public void setVille(String ville) {
        this.ville = ville;
    }

    public void setRue(String rue) {
        this.rue = rue;
    }

    public void setDate_commande(String date_commande) {
        this.date_commande = date_commande;
    }

   
    

    

    @Override
    public String toString() {
        return "Commande{" + "id=" + id + ", code_postal=" + code_postal + ", panier=" + panier + ", montant=" + montant + ", etat_commande=" + etat_commande + ", gouvernorat=" + gouvernorat + ", ville=" + ville + ", rue=" + rue + ", date_commande=" + date_commande + '}';
    }
    
    
}

File: src/Entity/Commande.php
<?php

namespace App\Entity;

use App\Repository\CommandeRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: CommandeRepository::class)]
class Commande
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name:"id_commande")]
    private ?int $id = null;

    #[ORM\ManyToOne]
    #[ORM\JoinColumn(nullable: false,name:"id_panier",referencedColumnName:"id_panier")]
    private ?Panier $panier = null;

    #[ORM\Column]
    private ?float $montant = null;

    #[ORM\Column(length: 200)]
    private ?string $etat_commande = null;

    #[ORM\Column(length: 200)]
    #[Assert\NotBlank(message:"Le nom du gouvernorat est obligatoire !")]
    private ?string $gouvernorat = null;

    #[ORM\Column(length: 100)]
    #[Assert\NotBlank(message:"Le nom de la ville est obligatoire !")]
    private ?string $ville = null;

    #[ORM\Column(length: 200)]
    #[Assert\NotBlank(message:"Le nom de la rue est obligatoire !")]
    private ?string $rue = null;

    #[ORM\Column]
    #[Assert\NotBlank(message:"Le code postal est obligatoire !")]
    #[Assert\Positive(message:"Le code postal doit etre un nombre positif !")]
    #[Assert\Length(exactly: 4, exactMessage : "Le code postal doit etre un nombre de quatre chiffres")]
    private ?int $code_postal = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    private ?\DateTimeInterface $date_commande  ;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getPanier(): ?Panier
    {
        return $this->panier;
    }

    public function setPanier(?Panier $panier): self
    {
        $this->panier = $panier;

        return $this;
    }

    public function getMontant(): ?float
    {
        return $this->montant;
    }

    public function setMontant(float $montant): self
    {
        $this->montant = $montant;

        return $this;
    }

    public function getEtatCommande(): ?string
    {
        return $this->etat_commande;
    }

    public function setEtatCommande(string $etat_commande): self
    {
        $this->etat_commande = $etat_commande;

        return $this;
    }

    public function getGouvernorat(): ?string
    {
        return $this->gouvernorat;
    }

    public function setGouvernorat(string $gouvernorat): self
    {
        $this->gouvernorat = $gouvernorat;

        return $this;
    }

    public function getVille(): ?string
    {
        return $this->ville;
    }

    public function setVille(string $ville): self
    {
        $this->ville = $ville;

        return $this;
    }

    public function getRue(): ?string
    {
        return $this->rue;
    }

    public function setRue(string $rue): self
    {
        $this->rue = $rue;

        return $this;
    }

    public function getCodePostal(): ?int
    {
        return $this->code_postal;
    }

    public function setCodePostal(int $code_postal): self
    {
        $this->code_postal = $code_postal;

        return $this;
    }

    public function getDateCommande(): ?\DateTimeInterface
    {
        return $this->date_commande;
    }

    public function setDateCommande(\DateTimeInterface $date_commande): self
    {
        $this->date_commande = $date_commande;

        return $this;
    }
}
